### PR TITLE
rust/dns: add v1 dns logging - v3

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/3569

Changes from last PR:
- Fix Lua support functions to match that of C, allowing the dns-lua suricata-verify test to pass.

Redmine issues:
- https://redmine.openinfosecfoundation.org/issues/2730
- https://redmine.openinfosecfoundation.org/issues/2704

Suricata-Verify has been updated as well:
- https://github.com/OISF/suricata-verify/pull/4

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/335
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/689

### CODE REVIEW

File: rust/src/dns/lua.rs [hunk 1]
```diff
@@ -93,11 +88,26 @@ pub extern "C" fn rs_dns_lua_get_query_table(clua: &mut CLuaState,
 
             lua.settable(-3);
         }
+    } else if let Some(response) = &tx.response {
```
Comment (victorjulien) [inline rust/src/dns/lua.rs]: is it possible to have both no request and no response? In this case returning 1 at the end looks suspicious. Though a table is created unconditionally.

File: rust/src/dns/log.rs [hunk 2]
```diff
@@ -598,3 +598,122 @@ pub extern "C" fn rs_dns_log_json_answer(tx: &mut DNSTransaction,
...
+}
+
+#[no_mangle]
+pub extern "C" fn rs_dns_log_json_answer_v1(tx: &mut DNSTransaction,
+                                         i: libc::uint16_t,
+                                         flags: libc::uint64_t)
+                                         -> *mut JsonT
+{
+    let index = i as usize;
+    for response in &tx.response {
```
[the remark this answers is not in the document]
Comment (jasonish) [reply rust/src/dns/log.rs]: Its on my todo list to remove using Options as iterators, but I didn't think an idiomatic change belonged in a 0.0.1 release in an effort to minimize the amount of code changed.  The v1 stuff was mostly a revert of a commit, so carried that back in with it. But its consistent with other stuff in dns/log.rs.  Happy to make the change though.

[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/dns/log.rs]: I prefer to have it fixed now before myself or anyone else spends time being confused again :)

File: rust/src/dns/lua.rs (hunk 1 above)
Comment (jasonish) [reply rust/src/dns/lua.rs]: Yes, but this is for consistency with C, which always returns 1 and an empty table as well.

Comment (victorjulien) [reply rust/src/dns/lua.rs]: Ok. There should probably be a comment about this behavior in the code then.

**Comment by jasonish** [comment]
Closing. See https://github.com/OISF/suricata/pull/3574 for update.

**Pull request closed without merging**